src/main/java/eli/parser/Parser.java

```diff
@@ -0,0 +1,95 @@
+package eli.parser;
+
+import eli.command.*;
+import eli.exception.*;
+import eli.task.*;
+
+/**
+ * Deals with making sense of the user command.
+ */
+public class Parser {
+
+  /**
+   * Parses user input into a Command object.
+   *
+   * @param fullCommand The full input command string.
+   * @return The Command object representing the user's command.
+   * @throws EliException If the command is invalid or malformed.
+   */
+  public static Command parse(String fullCommand) throws EliException {
+    String[] parts = fullCommand.split(" ", 2);
+    String commandWord = parts[0];
+    String description = parts.length > 1 ? parts[1].trim() : "";
+
+    switch (commandWord.toLowerCase()) {
+      case "list":
+        return new ListCommand();
+      case "mark":
+        return new MarkCommand(Integer.parseInt(description));
+      case "unmark":
+        return new UnmarkCommand(Integer.parseInt(description));
+      case "delete":
+        return new DeleteCommand(Integer.parseInt(description));
+      case "todo":
+        if (description.isEmpty()) {
+          throw new EliException("The description of a todo cannot be empty.");
+        }
+        return new AddCommand(new ToDo(description));
+      case "deadline":
+        String[] deadlineParts = description.split(" /by ");
+        if (deadlineParts.length < 2) {
+          throw new EliException("The deadline must be in the format 'description /by yyyy-MM-dd'.");
+        }
+        return new AddCommand(new Deadline(deadlineParts[0], deadlineParts[1]));
+      case "event":
+        String[] eventParts = description.split(" /from ");
+        if (eventParts.length < 2) {
+          throw new EliException("The event must be in the format 'description /from yyyy-MM-dd /to yyyy-MM-dd'.");
+        }
+        String[] times = eventParts[1].split(" /to ");
+        return new AddCommand(new Event(eventParts[0], times[0], times[1]));
+      case "find":
+        if (description.isEmpty()) {
+          throw new EliException("The keyword for finding tasks cannot be empty.");
+        }
+        return new FindCommand(description);
+      case "bye":
+        return new ExitCommand();
+      default:
+        throw new UnknownCommandException();
+    }
```

Comment:
Based on our coding standard, the indentation for `case` should be on the same level as `switch`.